File: solutions/python/matrix/1/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix_rows = [[int(number) for number in row.split()] for row in matrix_string.split('\n')]
        self.matrix_columns = [[row[i] for row in self.matrix_rows] for i in range(0, len(self.matrix_rows[0]))]

    def row(self, index):
        return self.matrix_rows[index-1]

    def column(self, index):
        return self.matrix_columns[index-1]

File: solutions/javascript/matrix/1/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  constructor(stringMatrix) {
    const stringRows = stringMatrix.split("\n");
    this._rows = [];
    this._columns = [];
    
    stringRows.forEach((row, index) => {
      row = row.split(" ");
      row = row.map((element, index) => {
        const intElement = parseInt(element);
        if(!this._columns[index])
          this._columns[index] = [];
        this._columns[index].push(intElement);
        return intElement;
      });
      this._rows.push(row);
    })
  }

  get rows() {
    return this._rows;
  }

  get columns() {
    return this._columns;
  }
}
